cupy/_manipulation/kind.py

```diff
@@ -2,6 +2,31 @@
 from cupy import _core
 
 
+def asarray_chkfinite(a, dtype=None, order=None):
+    """
+    Convert the given input to an array,
+    and raises an error if the input is NaNs or Infs.
+
+    Args:
+        a: array like.
+        dtype: data type, optional
+        order: {'C', 'F'}, optional
```

Comment:
It seems 'A' and 'K' are also supported.
https://numpy.org/doc/stable/reference/generated/numpy.asarray_chkfinite.html